src/bigdecimal.ts

```diff
@@ -4350,6 +4350,37 @@ export class BigDecimal {
             return BigDecimal.divideAndRound3(dividend, scaledDivisor, scale, roundingMode, scale);
         }
     }
+
+    /**
+     * Function called in case BigDecimal is implicitly converted to a primitive value.
+     * The function will throw when the code try to convert the BigDecimal value in the following cases:
+     * - Using a greater or lower comparison operators (>, <, <=, >=)
+     * - Using loose equality operators against number, string, symbol or bigint (==, !=)
+     * - Using an arithmetic operators (+, -, /, *, %, **, --, ++)
+     * - Using bitwise operators (&, |, ^, ~, >>, >>>, <<)
```

Comment:
Is there a place in the mozilla documentation where all the possible type coercions are listed? In each operator's documentation, the conversions are explained but I wonder if there is a list of operations that can do a call to toPrimitive.